File: src/main/java/com/nao4j/subtrol/service/ServiceService.java
package com.nao4j.subtrol.service;

import com.nao4j.subtrol.dto.ShortService;
import org.springframework.security.access.annotation.Secured;

import java.util.Set;

import static com.nao4j.subtrol.security.Roles.ADMIN;
import static com.nao4j.subtrol.security.Roles.USER;

public interface ServiceService {

    @Secured({USER, ADMIN})
    Set<ShortService> getAllForUser(String userId);

    @Secured({USER, ADMIN})
    ShortService create(String userId, ShortService service);

    @Secured({USER, ADMIN})
    ShortService remove(String userId, ShortService service);

}

File: src/main/kotlin/com/nao4j/subtrol/service/ServiceService.kt
package com.nao4j.subtrol.service

import com.nao4j.subtrol.dto.ShortService
import com.nao4j.subtrol.security.ADMIN
import com.nao4j.subtrol.security.USER
import org.springframework.security.access.annotation.Secured

interface ServiceService {

    @Secured(value = [USER, ADMIN])
    fun getAllForUser(userId: String): Set<ShortService>

    @Secured(value = [USER, ADMIN])
    fun create(userId: String, service: ShortService): ShortService

    @Secured(value = [USER, ADMIN])
    fun remove(userId: String, service: ShortService): ShortService

}
